
package galaxy;

option cc_generic_services = true;

message RunTaskRequest {
    optional int64 task_id = 1;
    optional string task_name = 2;
    optional bytes task_raw = 3;
    optional string cmd_line = 4;
    optional double cpu_share = 5;
    optional int64 mem_share = 6;
    optional int32 job_replicate_num = 7;
    optional int32 task_offset = 8;
    optional int64 job_id = 9; // use to rebuild master
}

message RunTaskResponse {
    optional int32 status = 1;
}

message KillTaskRequest {
    optional int64 task_id = 2;
}

message KillTaskResponse {
    optional int32 status = 2;
    optional string gc_path = 3;
}

service Agent {
    rpc RunTask(RunTaskRequest) returns(RunTaskResponse);
    rpc KillTask(KillTaskRequest) returns(KillTaskResponse);
}

